pkg/cron_workflow.go

```diff
@@ -668,3 +650,50 @@ func cronWorkflowColumns(extraColumns ...string) []string {
 
 	return results
 }
+
+func (c *Client) SelectCronWorkflowWithExtraColumns(namespace, uid string, extraColumns ...string) (error, *CronWorkflow) {
+	query, args, err := sb.Select(cronWorkflowColumns(extraColumns...)...).
+		From("cron_workflows cw").
+		Join("workflow_template_versions wtv ON wtv.id = cw.workflow_template_version_id").
+		Join("workflow_templates wt ON wt.id = wtv.workflow_template_id").
+		Where(sq.Eq{
+			"wt.namespace": namespace,
+			"cw.name":      uid,
+		}).
+		ToSql()
+
+	if err != nil {
+		return err, nil
+	}
+
+	cronWorkflow := &CronWorkflow{}
+	if err = c.DB.Get(cronWorkflow, query, args...); err != nil {
+		return err, nil
+	}
+
+	return nil, cronWorkflow
+}
+
+func (c *Client) DeleteCronWorkflowDb(namespace, uid string) error {
+	query := `DELETE FROM cron_workflows 
```

Comment:
Can this query be rewritten using Squirrel?

Reply:
Not with `using`.

But we can do a GET to load the cron workflow and then delete by id.

Reply:
Got it, for some reason I thought Squirrel allowed custom SQL to be mixed with their methods.

Reply:
If there isn't a cleaner way to do the `where` values, then this is fine.

Reply:
Added a refactoring. The only thing I am unsure of is I'm not using namespace anymore. Just the uid of the cron workflow.
It worked when I tested locally, but I didn't test multiple cron workflows.

Reply:
I'd use the namespace too to be safe. We should use that in all queries.

Reply:
Cron Workflows do not have a namespace column in the db.
The Workflow Template that gets selected does.

I'm testing with multiple cron workflows and seeing some constraint violation issues, investigating.

Reply:
Ah, they definitely need to have `namespace`. Need to fix that.

Reply:
This branch should have the namespace added and working, along with refactoring.